generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/extensions/kFunctionExtensions.kt

```diff
@@ -16,12 +16,23 @@
 
 package com.expediagroup.graphql.generator.internal.extensions
 
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import kotlin.reflect.KFunction
 import kotlin.reflect.KParameter
 import kotlin.reflect.full.valueParameters
 
-internal fun KFunction<*>.getValidArguments(): List<KParameter> =
-    this.valueParameters
-        .filterNot { it.isGraphQLContext() }
-        .filterNot { it.isGraphQLIgnored() }
-        .filterNot { it.isDataFetchingEnvironment() }
+private val logger: Logger = LoggerFactory.getLogger("schemaGenerator")
+
+internal fun KFunction<*>.getValidArguments(): List<KParameter> {
+    return this.valueParameters.mapNotNull {
+        when {
```

Comment:
This will also essentially be logged on every usage of the context class. I would like to keep this code in the generator and not in the server.

We may need to just keep a state value of if we have logged yet so we only log a message once

Reply:
usage here is fine, it is invoked from `generateFunction`

Reply:
Your previous commit had the log happen on every execution of a request. This log will only happen at generation time, so either at start up or building of the schema. It will log for every usage still of the type but only once on each generation of the schema. 

I guess I was asking if we should only log once for the entire schema if you are using the context interface at all but this is fine as it would allow you to track the usages in logs too